httpx/dispatch/connection.py

```diff
@@ -59,7 +61,13 @@ def __init__(
             await self.connect(verify=verify, cert=cert, timeout=timeout)
 
         if self.h2_connection is not None:
-            response = await self.h2_connection.send(request, timeout=timeout)
+            try:
+                response = await self.h2_connection.send(request, timeout=timeout)
+            except NoAvailableStreamIDError:
+                await self.connect(verify=verify, cert=cert, timeout=timeout)
+                # since the h2_connection is renewed, NoAvailableStreamIDError
+                # should not occur again
+                response = await self.h2_connection.send(request, timeout=timeout)
```

Comment:
Apologies, I'm lost on how we could handle the exception there. We would need to create a new connection if stream ids are not available. And since `H2Connections.send_headers` is inside `H2Connection` class, how would we create a new connection object?

Reply:
Yes, I see, don't know what I was thinking there. :)

The question I have then is — the spec encourages us to open a new connection, but what do we do about the existing one? Surely we can't just throw it away by replacing `h2_connection` with another one, and we can't close it because it might be in use by other requests/responses.

Reply:
I think according to the spec, since it doesn't mention anything about what should happen to the old spec, we should just let it be, and let the user still have access to it somehow. I might be wrong here, but since I can still get responses on stream ids from the old connection, throwing it away might be bad. It'll still be available in the `ConnectionPool` right?